tests/unit/components/sl-input-test.js

```diff
@@ -0,0 +1,210 @@
+import ember from 'ember';
+import { moduleForComponent, test } from 'ember-qunit';
+
+moduleForComponent( 'sl-input', 'Unit | Component | sl input', {
+    unit: true
+});
+
+
+test( 'Blur action is triggered when input loses focus', function( assert ) {
+    assert.expect( 1 );
+
+    const done = assert.async();
+
+    this.subject({
+        'blur': 'blur',
+        targetObject: {
+            blur() {
+                assert.ok(
+                    'blur was triggered'
+                );
+
+                done();
+            }
+        }
+    });
+
+    this.$( 'input' ).trigger( 'blur' );
+});
+
+test( 'Click to edit input has the correct class', function( assert ) {
+    this.subject({
+        clickToEdit: true
+    });
+
+    assert.ok(
+        this.$( 'input' ).hasClass( 'click-to-edit' )
+    );
+});
+
+test( 'Input can be disabled' , function( assert ) {
+    this.subject({
+        disabled: true
+    });
+
+    assert.ok(
+        this.$( 'input' ).prop( 'disabled' )
+    );
+});
+
+test( 'Help text is displayed', function( assert) {
+    const helpText = 'Testing help text is displayed';
+
+    this.subject({
+        helpText: helpText
+    });
+
+    assert.equal(
+        this.$( '.help-block' ).text().trim(),
+        helpText
+    );
+});
+
+test( 'Label text is displayed', function( assert ) {
```

Comment:
There should be a negation test to this one.  if the `{{#if label}}` condition in the template is removed this test will still pass, when what we are really wanting to assert is that the label is only rendered when the `label` property is populated.